src/gui/labeling/qgslabelinggui.cpp

```diff
@@ -913,10 +945,12 @@ void QgsLabelingGui::validateGeometryGeneratorExpression()
 {
   bool valid = true;
 
+  QgsVectorLayer *vLayer = qobject_cast< QgsVectorLayer * >( mLayer );
```

Comment:
```suggestion
  QgsVectorLayer *vLayer = qobject_cast< QgsVectorLayer * >( mLayer );
  if ( !vLayer )
    return;
```

Reply:
Same here, this would break validation of the geometry generator expression.